service/lib/agama/dbus/storage/manager.rb

```diff
@@ -132,26 +132,38 @@ def result
           dbus_proposal&.path || ::DBus::ObjectPath.new("/")
         end
 
+        # Default volume used as template
+        #
+        # @return [Hash]
+        def default_volume(mount_path)
+          volume = volume_templates_builder.for(mount_path)
+          VolumeConversion.to_dbus(volume)
+        end
+
         # Calculates a new proposal
         #
         # @param dbus_settings [Hash]
         # @return [Integer] 0 success; 1 error
         def calculate_proposal(dbus_settings)
           logger.info("Calculating storage proposal from D-Bus settings: #{dbus_settings}")
 
-          converter = ProposalSettingsConverter.new
-          success = proposal.calculate(converter.to_agama(dbus_settings))
+          settings = ProposalSettingsConversion.from_dbus(dbus_settings, config: config)
```

Comment:
I would probably also log here resulting settings so in logs we see dbus one and converted one for easier debugging.